proxy/http2/Http2ConnectionState.cc

```diff
@@ -180,12 +180,14 @@ rcv_data_frame(Http2ConnectionState &cstate, const Http2Frame &frame)
     if (nbytes + read_len > unpadded_length) {
       read_len -= nbytes + read_len - unpadded_length;
     }
-    nbytes += writer->write(myreader, read_len);
-    myreader->consume(nbytes);
+    int num_written = writer->write(myreader, read_len);
+    nbytes += num_written;
+    myreader->consume(num_written);
+    stream->update_read(num_written);
```

Comment:
It looks odd because the data is written to the `Http2Stream::read_vio.buffer` from the buffer of the frame.

- `IOBufferReader *myreader = frame.reader()->clone();`  (line 172)
- `MIOBuffer *writer = stream->read_vio_writer();` (line 164)

IIUC, incrementing `Http2Stream::read_vio.nbytes` make sense, but `Http2Stream::read_vio.ndone` should be incremented when the `Http2Stream::read_vio.buffer` is consumed.

Reply:
I completely misunderstood when the `vio.ndone` needs to be incremented.
In the UnixNetVConnection & SSLNetVConnection, it is incremented when the `vio.buffer` is written.
https://github.com/apache/trafficserver/blob/f78c723a6e54accad8b3c82a58900314bcdd5cf9/iocore/net/UnixNetVConnection.cc#L294
https://github.com/apache/trafficserver/blob/f78c723a6e54accad8b3c82a58900314bcdd5cf9/iocore/net/SSLNetVConnection.cc#L339